### PR TITLE
LPS-185207 Make className a partitioned control table

### PR DESCRIPTION
**Pull request by LuisOrtiz89**
Hi @shuyangzhou,
This is the revision of [this](https://github.com/shuyangzhou/liferay-portal/pull/11166) previous PR. 
In this [other](https://github.com/liferay-uniform/liferay-portal/pull/1359) PR our QA team approved the changes.
Please review it when you have time. We removed all the company browsing from the service and did it outside when needed.
Regards,
Luis

### CODE REVIEW

File: copyright.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
 /**
  * SPDX-FileCopyrightText: (c) 2023 Liferay, Inc. https://liferay.com
  * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
- */
\ No newline at end of file
+ */
```
Comment (shuyangzhou) [inline copyright.txt:4]: @LuisOrtiz89 what is this? Please remove this change.

**Comment by shuyangzhou** [comment]
@GinsonRen this should have a very minimum runtime performance impact, but let's be safe, run a few test cases to confirm.

**Comment by GinsonRen** [comment]
@111Anna Please run a diff on this, thanks.

**Comment by 111Anna** [comment]
Hi @shuyangzhou, I ran the PR on main1, there seems to have an obvious regression on  MB

 Here is the diff result,:

Login:
Portal Version | GitId |Session Count | Mean Time of Login | Session Count Change (%)
-- | -- | -- | -- | --
Baseline | 39b0896 |48308 | 263| -
Baseline + PR-11208 | - | 48212 |261 | -0.20%

WC:
Portal Version | GitId |Session Count | Mean Time of Login | Session Count Change (%)
-- | -- | -- | -- | --
Baseline | 39b0896 |64011 | 441| -
Baseline + PR-11208 | -|62370| 471 | -2.56% 

MB:
Portal Version | GitId |Session Count | Mean Time of Login | Session Count Change (%)
-- | -- | -- | -- | --
Baseline | 39b0896 |10962 | 100| -
Baseline + PR-11208 |-| 9884| 329 | -9.83% 

Blog:
Portal Version | GitId |Session Count | Mean Time of Login | Session Count Change (%)
-- | -- | -- | -- | --
Baseline | 39b0896 |17041 | 44.6| -
Baseline + PR-11208 |-| 17033| 42.7 | -0.04% 

DL:
Portal Version | GitId |Session Count | Mean Time of Login | Session Count Change (%)
-- | -- | -- | -- | --
Baseline | 39b0896 |32007 | 455| -
Baseline + PR-11208 | -|32747| 329 | 2.31% 

AP:
Portal Version | GitId |Session Count | Mean Time of Login | Session Count Change (%)
-- | -- | -- | -- | --
Baseline | 39b0896 |29324 | 408| -
Baseline + PR-11208 |-| 29253| 395 | -0.24%

**Comment by shuyangzhou** [comment]
@111Anna yeah, that's too much, can you do some profile for MB and WC and share them with @LuisOrtiz89 ?

**Comment by 111Anna** [comment]
OK, I ran the profiles and uploaded it to [here](https://drive.google.com/drive/u/0/folders/1hcPH89XxcTUWJYFvR_OEy8MWLnAY_Y4V).
> @111Anna yeah, that's too much, can you do some profile for MB and WC and share them with @LuisOrtiz89 ?

/cc @LuisOrtiz89 @GinsonRen

**Comment by LuisOrtiz89** [comment]
> OK, I ran the profiles and uploaded it to [here](https://drive.google.com/drive/u/0/folders/1hcPH89XxcTUWJYFvR_OEy8MWLnAY_Y4V).
> 
> > @111Anna yeah, that's too much, can you do some profile for MB and WC and share them with @LuisOrtiz89 ?
> 
> /cc @LuisOrtiz89 @GinsonRen

Thanks @111Anna, I will take a look at them.

**Comment by 111Anna** [comment]
Hi @LuisOrtiz89, There was an issue prevented us from generating new data on the latest master, which has been resolved today. And this test is already running, we will update results tomorrow.

**Comment by LuisOrtiz89** [comment]
Thanks @111Anna.

@shuyangzhou, let's wait for the new test run, but analyzing the method list from the profile, we think the issue is not related to our PR and some other commits must have been included in the results due to the difference in the master timeline between repositories.
We see that the method that increases the time a lot is `com.liferay.portal.template.freemarker.internal.FreeMarkerTemplate$$Lambda$1323.call()`, which in the baseline was being called 0 times, and now it is being called 1257 times. We can see from other method calls that it is related to the messages, so it seems related to a change in the Message Board itself.
Regarding the ClassName, the outer method call we see is `com.liferay.portal.service.impl.ClassNameLocalServiceImpl.getClassNameId(String)`, which is being called a similar amount of times than before, but now it takes 838ms more to complete all the calls, which is not too much.

cc/ @achaparro

**Comment by 111Anna** [comment]
Hi @LuisOrtiz89, In our last round of testing, the baseline and PR had different process counts on MB. For this test I double checked all the settings to make sure it was all the same. 
  
Here are test results:
   
   
   Login:
Portal Version | GitId |Session Count | Mean Time of Login | Session Count Change (%)
-- | -- | -- | -- | --
Baseline | 7b59b646 |48715 | 266| -
Baseline + PR-11208 | - | 48635 |257 | -0.16%

WC:
Portal Version | GitId |Session Count | Mean Time of Login | Session Count Change (%)
-- | -- | -- | -- | --
Baseline | 7b59b646 |62293 | 524| -
Baseline + PR-11208 | -|62948| 446 | 1.05%

MB:
Portal Version | GitId |Session Count | Mean Time of Login | Session Count Change (%)
-- | -- | -- | -- | --
Baseline | 7b59b646 |10926 | 411| -
Baseline + PR-11208 |-| 10673| 692 | -2.32%

Blog:
Portal Version | GitId |Session Count | Mean Time of Login | Session Count Change (%)
-- | -- | -- | -- | --
Baseline | 7b59b646 |17023 | 43.6| -
Baseline + PR-11208 |-| 17032| 43.1 | 0.05%

DL:
Portal Version | GitId |Session Count | Mean Time of Login | Session Count Change (%)
-- | -- | -- | -- | --
Baseline | 7b59b646 |32385 | 385| -
Baseline + PR-11208 | -|32503| 313 | 0.36%

AP:
Portal Version | GitId |Session Count | Mean Time of Login | Session Count Change (%)
-- | -- | -- | -- | --
Baseline | 7b59b646 |28966 | 414| -
Baseline + PR-11208 |-| 29195| 403 | 0.78%

**Comment by LuisOrtiz89** [comment]
Hi @111Anna, thanks for the tests. We have been talking with @shuyangzhou, and we have seen that the session count change the first time was -9,83% and now it is -2,32% without introducing any change in the code. Would it be possible to execute the MB tests one or two more times to make sure the results are consistent?

Do you know if the database used for the tests is an already existing database or if the tests use a database created on portal startup?

cc/ @achaparro

**Comment by LuisOrtiz89** [comment]
Thanks @111Anna!

@shuyangzhou How do you think we should proceed? From the last test runs it seems the performance is pretty stable. The session count always decreases, but just a little, and the mean time of login is highly variable, sometimes it is bigger and others is lower.
Do you think we could move forward the PR, or we should send some more scoped PR's as talked yesterday? We don't see clearly how to divide the PR in smaller pieces as it only touches ClassNames, but we could try something if needed.

**Comment by shuyangzhou** [comment]
Yes, I am letting it pass now https://github.com/brianchandotcom/liferay-portal/pull/140419

**Comment by LuisOrtiz89** [comment]
Thanks @shuyangzhou !!

**Pull request closed without merging**